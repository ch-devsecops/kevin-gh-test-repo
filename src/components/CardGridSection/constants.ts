const CAROUSEL_SCROLL_WIDTH = 271; // Item Width (259px) + Carousel X Padding (12px)

export default CAROUSEL_SCROLL_WIDTH;
