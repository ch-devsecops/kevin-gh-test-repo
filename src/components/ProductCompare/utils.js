import uniqBy from 'lodash/uniqBy';
import groupBy from 'lodash/groupBy';
import { flatten } from 'lodash/array';

import { pushGtmCtaClickAddToCompareEvent } from '../../utils/gtmEvents';
import { MARINE_SITE_NAME } from '../../utils/constants';

/**
 * Parses the given specifications and groups them by the proper trim id.
 * @param {array} specifications - API response with all specifications for the different trims.
 * @param {array} trims - The currently active trims (id and name).
 * @returns {Object} EngineData - The parsed information for the given engine id.
 * @returns {array} EngineData.labels - The labels for each specification.
 * @returns {array} EngineData.specsLabels - The values for each specification.
 */
export const getEnginesGroupedBySpecifications = (specifications = [], trims = []) => {
  if (!specifications?.length || !trims?.length) return {};
  const labels = uniqBy(specifications, 'displayOrder')
    ?.map(category => {
      const specsLabels = category?.specs?.map(s => ({
        name: s.name,
        label: s.label,
        displayOrder: s.displayOrder,
      }));
      return specsLabels;
    })
    ?.flat(1);

  const specificationByTrims = trims?.map(trim => {
    const trimId = trim?.detIdentifier;
    const trimSpecs = uniqBy(
      specifications.filter(spec => spec?.trimId?.toString() === trimId),
      'displayOrder',
    )
      ?.map(spec => spec.specs)
      ?.flat(1);
    const specificationsByLabel = labels?.map(label => trimSpecs?.find(spec => spec.name === label.name) || label);

    return specificationsByLabel;
  });

  return {
    labels,
    specs: specificationByTrims,
  };
};

/**
 * Returns a formatted product list data shape based on the site name.
 * @param {array} products - product catalog data from sitecore.
 * @param {string} appName - site name from sitcoreContext.
 * @returns {array} formattedProduct - Formatted products with useable data shape based on the catalog.
 */

/**
 * Sort data by exactly the detIds
 * @param {array} detIds - Set of DET IDs.
 * @param data
 * @returns {array} enginesList - List of engines found within sitecore catalog.
 */
export const sortDataByDetIds = (detIds, data) =>
  data.sort((a, b) => detIds.indexOf(a?.detIdentifier) - detIds.indexOf(b?.detIdentifier));

/**
 * Search models by detId to be found within sitecore catalog data.
 * @param {array} detIds - Set of DET IDs.
 * @param {array} sitecoreEngineList - Unrefined sitecore catalog list of engines.
 * @returns {array} enginesList - List of engines found within sitecore catalog.
 */
export const getDataFromCatalog = (detIds, sitecoreEngineList) => {
  if (detIds?.length <= 0) return [];

  const data = flatten(
    sitecoreEngineList?.map(series => series.productList?.filter(model => detIds.includes(model?.detIdentifier))),
  );
  return sortDataByDetIds(detIds, data);
};

/**
 * Gets the value of the given param from the URL.
 * @param {string} param The name of the parameter to be used/searched-for in the URL
 * @returns The values found.
 */
export const getParamsFromUrl = param => {
  const queryParams = new URLSearchParams(window.location.search);
  return (
    queryParams
      .get(param)
      ?.split(',')
      .filter(val => val) || undefined
  );
};

/**
 * Updates the URL and sets the given param to the given value.
 * @param {string} param The param to be used in the URL.
 * @param {Object} history History Object coming from react-router-dom.
 * @param {string[]} values The values to be set.
 */
export const updateSearchParams = (param, history, values) => {
  if (!Array.isArray(values) || typeof window === 'undefined') return;
  try {
    const valuesStr = values.toString();
    const queryParams = new URLSearchParams(window.location.search);
    queryParams.set(param, valuesStr);
    if (valuesStr) {
      const newUrl = `${window.location.pathname}?${decodeURIComponent(queryParams.toString())}`;
      history.replace(newUrl);
    } else {
      history.replace(window.location.pathname);
    }
  } catch (error) {
    console.warn('Product Compare: updateSearchParams error', error);
  }
};

/**
 * Pushes the gtmEvent based on the AppName.
 * @AppName {string} The name of the AppName used
 * @productId {string} The sitecore detModelKey
 * @category {string} The name category from sitecore
 * @returns The gtmEvent found.
 */
export const pushGtmViewDetailsCompareEvent = (AppName, productId, category) => {
  // TODO: Delete rule as soon as more apps are using this switch case
  // eslint-disable-next-line default-case
  switch (AppName) {
    case MARINE_SITE_NAME:
      return pushGtmCtaClickAddToCompareEvent(category, productId);
  }
};

/**
 * Parses the given specifications and groups them by the category and sub-category.
 * @param {Object[]} specifications - API response with all specifications for trims.
 */
export const getPSPCategorizedSpecifications = specifications => {
  if (!Array.isArray(specifications) || !specifications.length) return [];

  try {
    /* Group specs by the first layer of category name and transform to array to iterate over */
    const specsGroup = flatten(
      Object.entries(groupBy(specifications, 'name'))?.map(([categoryName, trimSpecs]) => {
        // Inject trimId into each spec item and merge/flatten specs from all trims
        const specsArrWithTrimId = flatten(
          trimSpecs.map(trimSpec => trimSpec?.specs?.map(spec => ({ ...spec, trimId: trimSpec?.trimId }))),
        );
        /* Group specs by the second layer of sub-category name and transform to array to iterate over */
        const trimSpecsGroupBySubCategory = flatten(
          Object.entries(groupBy(specsArrWithTrimId, 'name'))?.map(([subCategoryName, specs]) => ({
            name: subCategoryName,
            label: specs?.[0]?.label,
            specs,
          })),
        );
        return { name: categoryName, label: trimSpecs?.[0]?.label, specifications: trimSpecsGroupBySubCategory };
      }),
    );

    return specsGroup;
  } catch (error) {
    return undefined;
  }
};

export const getPspProductCardTitle = ({ year, model, transmission }) => {
  // filter out invalid values
  const title = [year, model, transmission].filter(value => !!value);
  // add space between
  return title.join(' ');
};
