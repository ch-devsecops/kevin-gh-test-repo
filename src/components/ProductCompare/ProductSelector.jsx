import React from 'react';
import PropTypes from 'prop-types';
import { nodeData, nodesData } from '@lukeaus/plain-tree';
import useAppName from '../../utils/sitecoreContext/useAppName';
import CascadingDropdown from './CascadingDropdown';
import NotImplemented from '../NotImplemented';
import { ENGINE_SITE_NAME, MARINE_SITE_NAME, PSP_SITE_NAME, PRODUCT_COMPARE_ID_KEY } from '../../utils/constants';
import { variant1, variant3 } from '../ProductCard/utils';

const ENGINE_DROPDOWN_SCHEMA = [
  {
    name: 'series',
    label: 'Shared.Common.productCategoryLabel',
    ariaLabel: 'Shared.Common.productCategoryAria',
  },
  {
    name: 'name',
    label: 'Shared.Common.productName',
    ariaLabel: 'Shared.Common.productNameAria',
  },
];

const PSP_DROPDOWN_SCHEMA = [
  {
    name: 'category',
    label: 'Shared.CompareTrims.categoryLabel',
    ariaLabel: 'category-aria',
  },
  {
    name: 'year',
    label: 'Shared.CompareTrims.yearLabel',
    ariaLabel: 'year-aria',
  },
  {
    name: 'model',
    label: 'Shared.CompareTrims.modelLabel',
    ariaLabel: 'model-aria',
  },
  {
    name: 'transmission',
    label: 'Shared.CompareTrims.transmissionLabel',
    ariaLabel: 'transmission-aria',
  },
];

const getEngineFilterOptions = (activeProducts, dropdownTotalLevels) => (optionNodes, currentLevel) => {
  const options = nodesData(optionNodes);
  if (!Array.isArray(options) || !options.length) return [];

  if (currentLevel === dropdownTotalLevels) {
    // filter out options already in activeProducts in the last level of dropdown
    return options.filter(option => !activeProducts.includes(option?.[PRODUCT_COMPARE_ID_KEY]));
  }
  return options;
};

const getPspFilterOptions = (activeProducts, dropdownTotalLevels) => (optionNodes, currentLevel) => {
  const options = nodesData(optionNodes);
  if (!Array.isArray(options) || !options.length) return [];

  if (currentLevel === dropdownTotalLevels - 1) {
    // filter out models that only have one transmission and that one also has already been selected
    const availableOptionNodes = optionNodes.filter(optionNode => {
      const childrenNodes = optionNode?.children || [];
      // if still any transmissions left unselected, keep it in options
      return childrenNodes?.some(childNode => !activeProducts.includes(nodeData(childNode)?.[PRODUCT_COMPARE_ID_KEY]));
    });
    return nodesData(availableOptionNodes);
  }

  if (currentLevel === dropdownTotalLevels) {
    // filter out options already in activeProducts in the last level of dropdown
    return options.filter(option => !activeProducts.includes(option?.[PRODUCT_COMPARE_ID_KEY]));
  }
  return options;
};

const ProductSelector = ({ productCatalogDataTree, onProductSelected, activeProducts, indexId }) => {
  const appName = useAppName();

  let variant;
  let dropdownSchema = [];
  let filterOptions = f => f;
  switch (appName) {
    case ENGINE_SITE_NAME:
    case MARINE_SITE_NAME:
      variant = variant1;
      dropdownSchema = ENGINE_DROPDOWN_SCHEMA;
      filterOptions = getEngineFilterOptions(activeProducts, dropdownSchema.length);
      break;

    case PSP_SITE_NAME:
      dropdownSchema = PSP_DROPDOWN_SCHEMA;
      filterOptions = getPspFilterOptions(activeProducts, dropdownSchema.length);
      variant = variant3;
      break;

    default:
      return <NotImplemented name="Product Compare: Product Selector" />;
  }

  return (
    <CascadingDropdown
      optionsTree={productCatalogDataTree}
      dropdownSchema={dropdownSchema}
      onCompleteSelection={onProductSelected}
      filterOptions={filterOptions}
      nameId={indexId}
      variant={variant}
    />
  );
};

ProductSelector.defaultProps = {
  indexId: 0,
};

ProductSelector.propTypes = {
  productCatalogDataTree: PropTypes.shape({}),
  activeProducts: PropTypes.arrayOf(PropTypes.string),
  onProductSelected: PropTypes.func,
  indexId: PropTypes.number,
};

export default ProductSelector;
