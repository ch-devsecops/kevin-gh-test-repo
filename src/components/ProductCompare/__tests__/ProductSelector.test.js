import 'jest-styled-components';
import React from 'react';
import { SitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import { axe, cleanup, jssRender } from 'test-utils';
import ProductSelector from '../ProductSelector';
import mockData from '../../../../sample-mock-data/routes/engine-compare/en.json';
import { ENGINE_SITE_NAME } from '../../../utils/constants';
import { getEngineMappedCatalogData } from '../../CompareProducts/utils';

const engineSeries = mockData?.fields?.data?.item?.series;

afterEach(cleanup);

const mockProps = {
  callback: jest.fn(),
  activeProducts: [],
  index: 0,
};

const sitecoreContextMock = {
  site: {
    name: ENGINE_SITE_NAME,
  },
  pageState: 'normal',
  hondaRestApiHost: 'uat-api.honda.ca',
};

describe('<ProductSelector>', () => {
  const catalogDataTree = getEngineMappedCatalogData(engineSeries);
  const renderMock = () =>
    jssRender(
      <SitecoreContext context={sitecoreContextMock} componentFactory={{}}>
        <ProductSelector
          productCatalogDataTree={catalogDataTree}
          selectorIndex={mockProps.index}
          activeProducts={mockProps.activeProducts}
          onProductSelected={mockProps.callback}
        />
      </SitecoreContext>,
    );

  it('should render without errors if there is no data', () => {
    const { container } = jssRender(
      <SitecoreContext context={sitecoreContextMock} componentFactory={{}}>
        <ProductSelector />
      </SitecoreContext>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should have no accessibility violations', async () => {
    const { container } = renderMock();

    expect(container).toMatchSnapshot();
    expect(await axe(container)).toHaveNoViolations();
  });

  it('renders without errors', async () => {
    const { container } = renderMock();

    expect(container).toMatchSnapshot();
  });
});
