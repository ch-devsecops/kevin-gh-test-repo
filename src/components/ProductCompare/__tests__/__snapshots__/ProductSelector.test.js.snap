// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductSelector> renders without errors 1`] = `
.c1 {
  width: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-width: 380px;
  margin: 0 auto;
  padding-top: 0;
  padding-left: xl;
  padding-right: xl;
}

.c2 {
  display: block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  color: #2C2C2C;
  font-family: Cooper Black;
  font-size: 12px;
  line-height: xs;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  font-family: Cooper Black;
  color: #2C2C2C;
}

.c4 {
  margin-bottom: l;
}

.c4 >div {
  background: #fff;
}

.c4 >div >span {
  text-overflow: ellipsis;
  word-break: break-word;
  white-space: nowrap;
  overflow: hidden;
}

.c4 >ul >li {
  white-space: normal;
  line-height: 24px;
}

.c3 {
  width: 100%;
  position: relative;
  cursor: pointer;
}

.c3:focus {
  outline: transparent;
}

.c3:focus div:first-of-type:after {
  height: 2px;
}

.c5 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  outline: none;
  width: 100%;
  font-size: 14px;
  color: #2C2C2C;
  font-family: Helvetica;
  font-weight: normal;
  line-height: 22px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  background-color: #F5F5F5;
  background-color: #fff;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: normal;
}

.c5::-ms-clear {
  display: none;
}

.c5::-webkit-input-placeholder {
  color: #707070;
}

.c5::-moz-placeholder {
  color: #707070;
}

.c5:-ms-input-placeholder {
  color: #707070;
}

.c5::placeholder {
  color: #707070;
}

.c5:after {
  content: '';
  position: absolute;
  width: 100%;
  bottom: 0;
  left: 0;
  height: 1px;
  background-color: #2C2C2C;
}

.c5:after {
  height: 1px;
}

.c6 {
  margin: 0;
  color: #2C2C2C;
  font-weight: normal;
  font-family: Helvetica;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  font-weight: normal;
  padding-left: 15px;
  padding-right: 15px;
  width: 100%;
  height: small;
}

.c8 {
  max-height: 0px;
  width: 100%;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  -webkit-transition-duration: t4;
  transition-duration: t4;
  position: absolute;
  top: 46px;
  overflow: hidden;
  outline: none;
  z-index: options;
}

.c8 li {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.c9 {
  font-size: 14px;
  padding: 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-text-size-adjust: 100%;
  background-color: #F8F8F8;
  color: #2C2C2C;
  font-family: Helvetica;
  font-weight: normal;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  border-bottom: 1px solid;
  border-color: #D5D5D5;
}

.c9:hover {
  background-color: #EAEAEA;
}

.c9.focusVisible {
  outline: undefined undefined;
}

.c9:hover {
  font-family: Cooper Black;
}

.c10 {
  display: block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  color: #2C2C2C;
  font-family: Cooper Black;
  font-size: 12px;
  line-height: xs;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  font-family: Cooper Black;
  color: #707070;
}

.c11 {
  width: 100%;
  position: relative;
  cursor: pointer;
  pointer-events: none;
}

.c11:focus {
  outline: transparent;
}

.c11:focus div:first-of-type:after {
  height: 2px;
}

.c12 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  outline: none;
  width: 100%;
  font-size: 14px;
  color: #2C2C2C;
  font-family: Helvetica;
  font-weight: normal;
  line-height: 22px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  background-color: #F5F5F5;
  background-color: #fff;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: normal;
}

.c12::-ms-clear {
  display: none;
}

.c12::-webkit-input-placeholder {
  color: #707070;
}

.c12::-moz-placeholder {
  color: #707070;
}

.c12:-ms-input-placeholder {
  color: #707070;
}

.c12::placeholder {
  color: #707070;
}

.c12:after {
  content: '';
  position: absolute;
  width: 100%;
  bottom: 0;
  left: 0;
  height: 1px;
  background-color: #2C2C2C;
}

.c12:after {
  height: 1px;
}

.c12:after {
  background-color: #C7C7C7;
}

.c12 svg path {
  stroke: #C7C7C7;
}

.c13 {
  margin: 0;
  color: #2C2C2C;
  font-weight: normal;
  font-family: Helvetica;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  font-weight: normal;
  padding-left: 15px;
  padding-right: 15px;
  width: 100%;
  height: small;
  color: #707070;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: iconWrapper.md;
  height: iconWrapper.md;
}

@media screen and (min-width:768px) {
  .c0 {
    padding-left: s;
    padding-right: s;
  }
}

@media screen and (min-width:768px) {
  .c2 {
    line-height: 22px;
  }
}

@media screen and (min-width:768px) {
  .c5 {
    line-height: 24px;
  }
}

@media screen and (min-width:768px) {
  .c10 {
    line-height: 22px;
  }
}

@media screen and (min-width:768px) {
  .c12 {
    line-height: 24px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      data-testid="product-series-wrapper"
      width="100%"
    >
      <label
        class="c2"
        data-testid="product-series-label"
      >
        Shared.Common.productCategoryLabel
      </label>
      <div
        aria-label="Shared.Common.productCategoryAria"
        class="c3 sc-1k44dfv-0 c4"
        data-testid="dsr-dropdown-container"
        tabindex="0"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          class="sc-1k44dfv-0 c5"
          data-testid="button"
          id="series-0"
          role="button"
        >
          <span
            class="sc-1k44dfv-0 c6"
            color="typographyDefault"
            data-testid="select-text"
            font-family="default"
            font-weight="regular"
          >
            Select
          </span>
          <div
            class="c7"
          />
        </div>
        <ul
          aria-label="Shared.Common.productCategoryAria"
          class="c8"
          role="listbox"
        >
          <li
            aria-selected="false"
            class="c9"
            data-testid="dsr-search-option"
            data-value="GX Series"
            role="option"
          >
            GX Series
          </li>
          <li
            aria-selected="false"
            class="c9"
            data-testid="dsr-search-option"
            data-value="Mini 4-Stroke"
            role="option"
          >
            Mini 4-Stroke
          </li>
          <li
            aria-selected="false"
            class="c9"
            data-testid="dsr-search-option"
            data-value="V-Twin Series"
            role="option"
          >
            V-Twin Series
          </li>
        </ul>
      </div>
    </div>
    <div
      class="c1"
      data-testid="product-name-wrapper"
      width="100%"
    >
      <label
        class="c10"
        data-testid="product-name-label"
      >
        Shared.Common.productName
      </label>
      <div
        aria-label="Shared.Common.productNameAria"
        class="c11 sc-1k44dfv-0 c4"
        data-testid="dsr-dropdown-container"
        disabled=""
        tabindex="0"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          class="sc-1k44dfv-0 c12"
          data-testid="button"
          disabled=""
          id="name-0"
          role="button"
        >
          <span
            class="sc-1k44dfv-0 c13"
            color="typographyDefault"
            data-testid="select-text"
            disabled=""
            font-family="default"
            font-weight="regular"
          >
            Select
          </span>
          <div
            class="c7"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ProductSelector> should have no accessibility violations 1`] = `
.c1 {
  width: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-width: 380px;
  margin: 0 auto;
  padding-top: 0;
  padding-left: xl;
  padding-right: xl;
}

.c2 {
  display: block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  color: #2C2C2C;
  font-family: Cooper Black;
  font-size: 12px;
  line-height: xs;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  font-family: Cooper Black;
  color: #2C2C2C;
}

.c4 {
  margin-bottom: l;
}

.c4 >div {
  background: #fff;
}

.c4 >div >span {
  text-overflow: ellipsis;
  word-break: break-word;
  white-space: nowrap;
  overflow: hidden;
}

.c4 >ul >li {
  white-space: normal;
  line-height: 24px;
}

.c3 {
  width: 100%;
  position: relative;
  cursor: pointer;
}

.c3:focus {
  outline: transparent;
}

.c3:focus div:first-of-type:after {
  height: 2px;
}

.c5 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  outline: none;
  width: 100%;
  font-size: 14px;
  color: #2C2C2C;
  font-family: Helvetica;
  font-weight: normal;
  line-height: 22px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  background-color: #F5F5F5;
  background-color: #fff;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: normal;
}

.c5::-ms-clear {
  display: none;
}

.c5::-webkit-input-placeholder {
  color: #707070;
}

.c5::-moz-placeholder {
  color: #707070;
}

.c5:-ms-input-placeholder {
  color: #707070;
}

.c5::placeholder {
  color: #707070;
}

.c5:after {
  content: '';
  position: absolute;
  width: 100%;
  bottom: 0;
  left: 0;
  height: 1px;
  background-color: #2C2C2C;
}

.c5:after {
  height: 1px;
}

.c6 {
  margin: 0;
  color: #2C2C2C;
  font-weight: normal;
  font-family: Helvetica;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  font-weight: normal;
  padding-left: 15px;
  padding-right: 15px;
  width: 100%;
  height: small;
}

.c8 {
  max-height: 0px;
  width: 100%;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  -webkit-transition-duration: t4;
  transition-duration: t4;
  position: absolute;
  top: 46px;
  overflow: hidden;
  outline: none;
  z-index: options;
}

.c8 li {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.c9 {
  font-size: 14px;
  padding: 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-text-size-adjust: 100%;
  background-color: #F8F8F8;
  color: #2C2C2C;
  font-family: Helvetica;
  font-weight: normal;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  border-bottom: 1px solid;
  border-color: #D5D5D5;
}

.c9:hover {
  background-color: #EAEAEA;
}

.c9.focusVisible {
  outline: undefined undefined;
}

.c9:hover {
  font-family: Cooper Black;
}

.c10 {
  display: block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  color: #2C2C2C;
  font-family: Cooper Black;
  font-size: 12px;
  line-height: xs;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  font-family: Cooper Black;
  color: #707070;
}

.c11 {
  width: 100%;
  position: relative;
  cursor: pointer;
  pointer-events: none;
}

.c11:focus {
  outline: transparent;
}

.c11:focus div:first-of-type:after {
  height: 2px;
}

.c12 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  outline: none;
  width: 100%;
  font-size: 14px;
  color: #2C2C2C;
  font-family: Helvetica;
  font-weight: normal;
  line-height: 22px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  background-color: #F5F5F5;
  background-color: #fff;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: normal;
}

.c12::-ms-clear {
  display: none;
}

.c12::-webkit-input-placeholder {
  color: #707070;
}

.c12::-moz-placeholder {
  color: #707070;
}

.c12:-ms-input-placeholder {
  color: #707070;
}

.c12::placeholder {
  color: #707070;
}

.c12:after {
  content: '';
  position: absolute;
  width: 100%;
  bottom: 0;
  left: 0;
  height: 1px;
  background-color: #2C2C2C;
}

.c12:after {
  height: 1px;
}

.c12:after {
  background-color: #C7C7C7;
}

.c12 svg path {
  stroke: #C7C7C7;
}

.c13 {
  margin: 0;
  color: #2C2C2C;
  font-weight: normal;
  font-family: Helvetica;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  font-weight: normal;
  padding-left: 15px;
  padding-right: 15px;
  width: 100%;
  height: small;
  color: #707070;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: iconWrapper.md;
  height: iconWrapper.md;
}

@media screen and (min-width:768px) {
  .c0 {
    padding-left: s;
    padding-right: s;
  }
}

@media screen and (min-width:768px) {
  .c2 {
    line-height: 22px;
  }
}

@media screen and (min-width:768px) {
  .c5 {
    line-height: 24px;
  }
}

@media screen and (min-width:768px) {
  .c10 {
    line-height: 22px;
  }
}

@media screen and (min-width:768px) {
  .c12 {
    line-height: 24px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      data-testid="product-series-wrapper"
      width="100%"
    >
      <label
        class="c2"
        data-testid="product-series-label"
      >
        Shared.Common.productCategoryLabel
      </label>
      <div
        aria-label="Shared.Common.productCategoryAria"
        class="c3 sc-1k44dfv-0 c4"
        data-testid="dsr-dropdown-container"
        tabindex="0"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          class="sc-1k44dfv-0 c5"
          data-testid="button"
          id="series-0"
          role="button"
        >
          <span
            class="sc-1k44dfv-0 c6"
            color="typographyDefault"
            data-testid="select-text"
            font-family="default"
            font-weight="regular"
          >
            Select
          </span>
          <div
            class="c7"
          />
        </div>
        <ul
          aria-label="Shared.Common.productCategoryAria"
          class="c8"
          role="listbox"
        >
          <li
            aria-selected="false"
            class="c9"
            data-testid="dsr-search-option"
            data-value="GX Series"
            role="option"
          >
            GX Series
          </li>
          <li
            aria-selected="false"
            class="c9"
            data-testid="dsr-search-option"
            data-value="Mini 4-Stroke"
            role="option"
          >
            Mini 4-Stroke
          </li>
          <li
            aria-selected="false"
            class="c9"
            data-testid="dsr-search-option"
            data-value="V-Twin Series"
            role="option"
          >
            V-Twin Series
          </li>
        </ul>
      </div>
    </div>
    <div
      class="c1"
      data-testid="product-name-wrapper"
      width="100%"
    >
      <label
        class="c10"
        data-testid="product-name-label"
      >
        Shared.Common.productName
      </label>
      <div
        aria-label="Shared.Common.productNameAria"
        class="c11 sc-1k44dfv-0 c4"
        data-testid="dsr-dropdown-container"
        disabled=""
        tabindex="0"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          class="sc-1k44dfv-0 c12"
          data-testid="button"
          disabled=""
          id="name-0"
          role="button"
        >
          <span
            class="sc-1k44dfv-0 c13"
            color="typographyDefault"
            data-testid="select-text"
            disabled=""
            font-family="default"
            font-weight="regular"
          >
            Select
          </span>
          <div
            class="c7"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ProductSelector> should render without errors if there is no data 1`] = `<div />`;
